packages/babel-helper-create-class-features-plugin/src/decorators.ts

```diff
@@ -646,38 +646,72 @@ function transformClass(
     classIdLocal: t.Identifier;
 
   const decoratorsThis = new Map<t.Decorator, t.Expression>();
-  const maybeExtractDecorator = (decorator: t.Decorator) => {
-    const { expression } = decorator;
-    if (version === "2023-05" && t.isMemberExpression(expression)) {
-      let object;
-      if (
-        t.isSuper(expression.object) ||
-        t.isThisExpression(expression.object)
-      ) {
-        object = memoiseExpression(t.thisExpression(), "obj");
-      } else if (!scopeParent.isStatic(expression.object)) {
-        object = memoiseExpression(expression.object, "obj");
-        expression.object = object;
-      } else {
-        object = expression.object;
+  const maybeExtractDecorators = (
+    decorators: t.Decorator[],
+    memoise = false,
```

Comment:
Nit: How about `memoiseInPlace`? And if `memoiseInPlace` is true, the return value of this function should be `false` since decorator expressions have already been memoised.

`maybeExtractDecorators` is used twice: one is in class decorator and the other is in element decorator. At this moment we could have avoided the default parameter and pass the `memoiseInPlace` explicitly.